ergotree-ir/src/mir/subst_const.rs

```diff
@@ -52,12 +52,18 @@ impl SigmaSerializable for SubstConstants {
         positions
             .check_post_eval_tpe(&SType::SColl(SType::SInt.into()))
             .map_err(InvalidArgumentError::from)?;
-        let new_values = Expr::sigma_parse(r)?.into();
-        Ok(Self {
-            script_bytes,
-            positions,
-            new_values,
-        })
+        let new_values = Box::new(Expr::sigma_parse(r)?);
+        match new_values.post_eval_tpe() {
```

Comment:
Good job checking parsed IR node! We usually make a checking constructor, see the one in `Coll.map` - https://github.com/ergoplatform/sigma-rust/blob/828f863eb1e0c11a0c033c8b1ba8b1e642cc6d33/ergotree-ir/src/mir/coll_map.rs#L26-L46

Reply:
Fixed